### ISSUE TITLE
Replacement functions are incorrectly marked as errors

### ISSUE THREAD

**Issue by stefaneng** [jimhester/lintr#791]
Replacement functions are not valid for many of the linters.

Using the example from Advanced R:
```{r}
`second<-` <- function(x, value) {
  x[2] <- value
  x
}
```
Should be an acceptable function name for camelCase, snake_case, dotted.case, lowercase.
